Always use local repo on install?

@mstormi 

After #961 if I am correct the installer will always use the local repo. This is caused because of this:
``` bash
# must not remove for offline to work
#if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
# shellcheck disable=SC2154
echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
if ! openhabian_update; then
  echo "$(timestamp) [openHABian] The git repository on the public internet is not reachable."
  echo "$(timestamp) [openHABian] We will continue trying to get your system installed, but this is not guaranteed to work."
fi
ln -sfn /opt/openhabian/openhabian-setup.sh /usr/local/bin/openhabian-config
```
This would work, but `openhabian_update` is defined in `functions/openhabian.bash` which as far as I can tell is never sourced or included in `first-boot.bash` which means that it will always fail and default to the local repo, but this is not the right thing to do because then we won't update to the latest available version when we install from an image.

Is my logic right?

--- a/.travis.yml
+++ b/.travis.yml
@@ -49,6 +49,7 @@ jobs:
       arch: amd64
       script:
         # 64bit Azul Java 8 on a 64bit amd64 virtual platform, running in a 64bit OS on Travis AMD64 HW
+        - sed -i 's/^debugmode=.*$/debugmode=maximum/' build-image/openhabian.conf
         - sed -i 's/^java_opt=.*$/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .
         - echo -e "\n\e[36mopenHABian test installation using 64bit Azul Enterprise Java 8 running on a\e[31m 64bit OS\e[36m running on native ${HW}...\e[0m\n"

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,7 +50,6 @@ jobs:
       script:
         # 64bit Azul Java 8 on a 64bit amd64 virtual platform, running in a 64bit OS on Travis AMD64 HW
         - sed -i 's/^debugmode=.*$/debugmode=maximum/' build-image/openhabian.conf
-        - sed -i 's/^repositoryurl=.*$/repositoryurl=https://github.com/openhab/openhabian.git/' build-image/openhabian.conf
         - sed -i 's/^clonebranch=.*$/clonebranch=stable/' build-image/openhabian.conf
         - sed -i 's/^java_opt=.*$/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .

--- a/.travis.yml
+++ b/.travis.yml
@@ -53,7 +53,7 @@ jobs:
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .
         - echo -e "\n\e[36mopenHABian test installation using 64bit Azul Enterprise Java 8 running on a\e[31m 64bit OS\e[36m running on native ${HW}...\e[0m\n"
         - docker run --name "openhabian-${HW}" -d openhabian/${HW}-openhabian;
-        - docker exec -i "openhabian-${HW}" bash -c './build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local'
+        - docker exec -i "openhabian-${HW}" bash -c './build.bash local-test && mv ~/.profile ~/.bash_profile && /boot/first-boot.bash'
     - name: openHABian installation test with 64 bit Azul Java 11 on native amd64 on Ubuntu
       env: HW=amd64
       arch: amd64

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,8 +19,8 @@ services:
   - docker
 
 stages:
-  - BATS and ShellCheck tests
   - openHABian installation tests
+  - BATS and ShellCheck tests
 
 install:
   - ./test.bash travis

--- a/.travis.yml
+++ b/.travis.yml
@@ -19,8 +19,8 @@ services:
   - docker
 
 stages:
-  - openHABian installation tests
   - BATS and ShellCheck tests
+  - openHABian installation tests
 
 install:
   - ./test.bash travis
@@ -49,7 +49,6 @@ jobs:
       arch: amd64
       script:
         # 64bit Azul Java 8 on a 64bit amd64 virtual platform, running in a 64bit OS on Travis AMD64 HW
-        - sed -i 's/^debugmode=.*$/debugmode=maximum/' build-image/openhabian.conf
         - sed -i 's/^java_opt=.*$/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .
         - echo -e "\n\e[36mopenHABian test installation using 64bit Azul Enterprise Java 8 running on a\e[31m 64bit OS\e[36m running on native ${HW}...\e[0m\n"

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,6 +50,7 @@ jobs:
       script:
         # 64bit Azul Java 8 on a 64bit amd64 virtual platform, running in a 64bit OS on Travis AMD64 HW
         - sed -i 's/^debugmode=.*$/debugmode=maximum/' build-image/openhabian.conf
+        - sed -i 's/^repositoryurl=.*$/repositoryurl=https://github.com/openhab/openhabian.git/' build-image/openhabian.conf
         - sed -i 's/^clonebranch=.*$/clonebranch=stable/' build-image/openhabian.conf
         - sed -i 's/^java_opt=.*$/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,6 +50,7 @@ jobs:
       script:
         # 64bit Azul Java 8 on a 64bit amd64 virtual platform, running in a 64bit OS on Travis AMD64 HW
         - sed -i 's/^debugmode=.*$/debugmode=maximum/' build-image/openhabian.conf
+        - sed -i 's/^clonebranch=.*$/clonebranch=stable/' build-image/openhabian.conf
         - sed -i 's/^java_opt=.*$/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .
         - echo -e "\n\e[36mopenHABian test installation using 64bit Azul Enterprise Java 8 running on a\e[31m 64bit OS\e[36m running on native ${HW}...\e[0m\n"

--- a/.travis.yml
+++ b/.travis.yml
@@ -50,7 +50,6 @@ jobs:
       script:
         # 64bit Azul Java 8 on a 64bit amd64 virtual platform, running in a 64bit OS on Travis AMD64 HW
         - sed -i 's/^debugmode=.*$/debugmode=maximum/' build-image/openhabian.conf
-        - sed -i 's/^clonebranch=.*$/clonebranch=stable/' build-image/openhabian.conf
         - sed -i 's/^java_opt=.*$/java_opt=Zulu8-64/' build-image/openhabian.conf
         - docker build -t openhabian/${HW}-openhabian -f Dockerfile.amd64 .
         - echo -e "\n\e[36mopenHABian test installation using 64bit Azul Enterprise Java 8 running on a\e[31m 64bit OS\e[36m running on native ${HW}...\e[0m\n"

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -31,7 +31,9 @@ sed -i 's/\r$//' "$CONFIGFILE"
 # shellcheck disable=SC1090
 source "$CONFIGFILE"
 # shellcheck disable=SC1091
-source "/boot/helpers.bash"; source "/boot/openhabian.bash"
+source "/boot/helpers.bash"
+# shellcheck disable=SC1091
+source "/boot/openhabian.bash"
 echo "OK"
 
 # shellcheck disable=SC2154

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -30,6 +30,8 @@ sed -i 's/\r$//' "$CONFIGFILE"
 
 # shellcheck disable=SC1090
 source "$CONFIGFILE"
+source "/boot/helpers.bash"
+source "/boot/openhabian.bash"
 echo "OK"
 
 # shellcheck disable=SC2154

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -194,7 +194,8 @@ if (/bin/bash /opt/openhabian/openhabian-setup.sh unattended); then
   rm -f /opt/openHABian-install-inprogress
   touch /opt/openHABian-install-successful
 else
-  fail_inprogress
+  echo "$(timestamp) [openHABian] We tried to get your system installed, but without proper internet connectivity this is not guaranteed to work."
+  #fail_inprogress
 fi
 echo "$(timestamp) [openHABian] Execution of 'openhabian-setup.sh unattended' completed."
 

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -182,7 +182,7 @@ fi
 #if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
 # shellcheck disable=SC2154
 echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
-git -C /opt/openhabian add origin https://github.com/openhab/openhabian.git
+git -C /opt/openhabian remote add origin https://github.com/openhab/openhabian.git
 if ! openhabian_update; then
   echo "$(timestamp) [openHABian] The git repository on the public internet is not reachable."
   echo "$(timestamp) [openHABian] We will continue trying to get your system installed, but this is not guaranteed to work."

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -182,8 +182,6 @@ fi
 #if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
 # shellcheck disable=SC2154
 echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
-git -C /opt/openhabian remote -v
-#git -C /opt/openhabian remote add origin https://github.com/openhab/openhabian.git
 if ! openhabian_update; then
   echo "$(timestamp) [openHABian] The git repository on the public internet is not reachable."
   echo "$(timestamp) [openHABian] We will continue trying to get your system installed, but this is not guaranteed to work."

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -178,8 +178,8 @@ else
   echo "OK"
 fi
 
-# must not remove for offline to work
-#if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
+# must not remove /opt/openhabian for offline to work
+if [ -d /opt/openhabian ]; then git clone https://github.com/openhab/openhabian.git /opt/openhabian; fi
 # shellcheck disable=SC2154
 echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
 if ! openhabian_update; then

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -30,8 +30,8 @@ sed -i 's/\r$//' "$CONFIGFILE"
 
 # shellcheck disable=SC1090
 source "$CONFIGFILE"
-source "/boot/helpers.bash"
-source "/boot/openhabian.bash"
+# shellcheck disable=SC1091
+source "/boot/helpers.bash"; source "/boot/openhabian.bash"
 echo "OK"
 
 # shellcheck disable=SC2154

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -182,6 +182,7 @@ fi
 #if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
 # shellcheck disable=SC2154
 echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
+git -C /opt/openhabian add origin https://github.com/openhab/openhabian.git
 if ! openhabian_update; then
   echo "$(timestamp) [openHABian] The git repository on the public internet is not reachable."
   echo "$(timestamp) [openHABian] We will continue trying to get your system installed, but this is not guaranteed to work."

--- a/build-image/first-boot.bash
+++ b/build-image/first-boot.bash
@@ -182,7 +182,8 @@ fi
 #if [ -d /opt/openhabian ]; then cd /opt && rm -rf /opt/openhabian; fi
 # shellcheck disable=SC2154
 echo -n "$(timestamp) [openHABian] Cloning myself from ${repositoryurl}, ${clonebranch} branch... "
-git -C /opt/openhabian remote add origin https://github.com/openhab/openhabian.git
+git -C /opt/openhabian remote -v
+#git -C /opt/openhabian remote add origin https://github.com/openhab/openhabian.git
 if ! openhabian_update; then
   echo "$(timestamp) [openHABian] The git repository on the public internet is not reachable."
   echo "$(timestamp) [openHABian] We will continue trying to get your system installed, but this is not guaranteed to work."

--- a/build.bash
+++ b/build.bash
@@ -154,7 +154,8 @@ elif [ "$1" == "local-test" ]; then
   ln -sf /etc/systemd/system/openhabian-installer.service /etc/systemd/system/multi-user.target.wants/openhabian-installer.service
   rm -f /opt/openHABian-install-successful
   rm -f /opt/openHABian-install-inprogress
-  sed -i -e '1r functions/helpers.bash' /boot/first-boot.bash # Add platform identification
+  sed -i -e '1r functions/helpers.bash' /boot/first-boot.bash
+  cp functions/helpers.bash functions/openhabian.bash /boot/  # add platform identification and update functions
   # Use local filesystem's version of openHABian
   sed -i 's|openhabian_update|true|' /boot/first-boot.bash
   chmod +x /boot/first-boot.bash

--- a/build.bash
+++ b/build.bash
@@ -154,10 +154,11 @@ elif [ "$1" == "local-test" ]; then
   ln -sf /etc/systemd/system/openhabian-installer.service /etc/systemd/system/multi-user.target.wants/openhabian-installer.service
   rm -f /opt/openHABian-install-successful
   rm -f /opt/openHABian-install-inprogress
-  sed -i -e '1r functions/helpers.bash' /boot/first-boot.bash
   cp functions/helpers.bash functions/openhabian.bash /boot/  # add platform identification and update functions
   # Use local filesystem's version of openHABian
-  sed -i 's|openhabian_update|true|' /boot/first-boot.bash
+  if ! running_in_docker; then
+    sed -i 's|openhabian_update|true|' /boot/first-boot.bash
+  fi
   chmod +x /boot/first-boot.bash
   chmod +x /boot/webif.bash
   echo_process "Local system ready for installation test. Run 'systemctl start openhabian-installer' or reboot to initiate"

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -84,7 +84,9 @@ openhabian_update_check() {
 
 openhabian_update() {
   local branch shorthash_before
-  current=$(git -C "${BASEDIR:-/opt/openhabian}" rev-parse --abbrev-ref HEAD)
+
+  export BASEDIR="${BASEDIR:-/opt/openhabian}"
+  current=$(git -C "${BASEDIR}" rev-parse --abbrev-ref HEAD)
   if [ "$current" == "master" ]; then
     local introtext="You are currently using the very latest (\"master\") version of openHABian.\\nThis is providing you with the latest features but less people have tested it so it is a little more likely that you run into errors.\\nWould you like to step back a little now and switch to use the stable version ?\\nYou can switch at any time by selecting this menu option again or by setting the clonebranch= parameter in /etc/openhabian.conf.\\n"
   else

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -120,6 +120,9 @@ openhabian_update() {
   else
     branch="${sel:-stable}"
   fi
+  if [[ -n "$UNATTENDED" ]]; then
+    branch=${clonebranch:-stable}
+  fi
 
   shorthash_before=$(git -C "$BASEDIR" log --pretty=format:'%h' -n 1)
   git -C "$BASEDIR" fetch --quiet origin || FAILED=1

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -100,7 +100,6 @@ openhabian_update() {
   fi
 
   FAILED=0
-  echo "unatt = $UNATTENDED"
   if [[ -n "$INTERACTIVE" ]]; then
     if [[ "$current" == "stable" || "$current" == "master" ]]; then
       if ! sel=$(whiptail --title "openHABian version" --radiolist "$introtext" 14 75 2 stable "recommended standard version of openHABian" on master "very latest version of openHABian" off 3>&1 1>&2 2>&3); then return 0; fi

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -98,6 +98,7 @@ openhabian_update() {
   fi
 
   FAILED=0
+  echo "unatt = $UNATTENDED"
   if [[ -n "$INTERACTIVE" ]]; then
     if [[ "$current" == "stable" || "$current" == "master" ]]; then
       if ! sel=$(whiptail --title "openHABian version" --radiolist "$introtext" 14 75 2 stable "recommended standard version of openHABian" on master "very latest version of openHABian" off 3>&1 1>&2 2>&3); then return 0; fi
@@ -105,22 +106,22 @@ openhabian_update() {
       if ! sel=$(whiptail --title "openHABian version" --radiolist "$introtext" 14 75 3 stable "recommended standard version of openHABian" off master "very latest version of openHABian" off "$current" "some other version you fetched yourself" on 3>&1 1>&2 2>&3); then return 0; fi
     fi
     sed -i "s@^clonebranch=.*@clonebranch=$sel@g" "/etc/openhabian.conf"
-  fi
-  echo -n "$(timestamp) [openHABian] Updating myself... "
-  read -r -t 1 -n 1 key
-  if [ "$key" != "" ]; then
-    echo -e "\\nRemote git branches available:"
-    git -C "$BASEDIR" branch -r
-    read -r -e -p "Please enter the branch to checkout: " branch
-    branch="${branch#origin/}"
-    if ! git -C "$BASEDIR" branch -r | grep -q "origin/$branch"; then
-      echo "FAILED - The custom branch does not exist."
-      return 1
+#  fi
+    echo -n "$(timestamp) [openHABian] Updating myself... "
+    read -r -t 1 -n 1 key
+    if [ "$key" != "" ]; then
+      echo -e "\\nRemote git branches available:"
+      git -C "$BASEDIR" branch -r
+      read -r -e -p "Please enter the branch to checkout: " branch
+      branch="${branch#origin/}"
+      if ! git -C "$BASEDIR" branch -r | grep -q "origin/$branch"; then
+        echo "FAILED - The custom branch does not exist."
+        return 1
+      fi
+    else
+      branch="${sel:-stable}"
     fi
   else
-    branch="${sel:-stable}"
-  fi
-  if [[ -n "$UNATTENDED" ]]; then
     branch=${clonebranch:-stable}
   fi
 

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -108,7 +108,6 @@ openhabian_update() {
       if ! sel=$(whiptail --title "openHABian version" --radiolist "$introtext" 14 75 3 stable "recommended standard version of openHABian" off master "very latest version of openHABian" off "$current" "some other version you fetched yourself" on 3>&1 1>&2 2>&3); then return 0; fi
     fi
     sed -i "s@^clonebranch=.*@clonebranch=$sel@g" "/etc/openhabian.conf"
-#  fi
     echo -n "$(timestamp) [openHABian] Updating myself... "
     read -r -t 1 -n 1 key
     if [ "$key" != "" ]; then

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -84,7 +84,7 @@ openhabian_update_check() {
 
 openhabian_update() {
   local branch shorthash_before
-  current=$(git -C "$BASEDIR" rev-parse --abbrev-ref HEAD)
+  current=$(git -C "${BASEDIR:-/opt/openhabian}" rev-parse --abbrev-ref HEAD)
   if [ "$current" == "master" ]; then
     local introtext="You are currently using the very latest (\"master\") version of openHABian.\\nThis is providing you with the latest features but less people have tested it so it is a little more likely that you run into errors.\\nWould you like to step back a little now and switch to use the stable version ?\\nYou can switch at any time by selecting this menu option again or by setting the clonebranch= parameter in /etc/openhabian.conf.\\n"
   else

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -31,6 +31,8 @@ openhabian_announcements() {
   local newsfile="${BASEDIR}/NEWS.md"
   local readnews="${BASEDIR}/docs/LASTNEWS.md"
 
+  if ! [[ -n "$INTERACTIVE" ]]; then return 1; fi
+
   if ! diff -q "$newsfile" "$readnews" >/dev/null 2>&1; then
     # shellcheck disable=SC2086
     if (whiptail --title "openHABian announcements" --yes-button "Stop Displaying" --no-button "Keep Displaying" --defaultno --scrolltext --yesno "$(cat $newsfile)" 27 85); then

--- a/functions/openhabian.bash
+++ b/functions/openhabian.bash
@@ -31,7 +31,7 @@ openhabian_announcements() {
   local newsfile="${BASEDIR}/NEWS.md"
   local readnews="${BASEDIR}/docs/LASTNEWS.md"
 
-  if ! [[ -n "$INTERACTIVE" ]]; then return 1; fi
+  if [[ -z "$INTERACTIVE" ]]; then return 1; fi
 
   if ! diff -q "$newsfile" "$readnews" >/dev/null 2>&1; then
     # shellcheck disable=SC2086

